### ISSUE TITLE
About the patients splits between train_data and domain_data

### ISSUE THREAD

**Issue by MohSamNaf** [ziyujia/MSTGCN#8]
Hi. Thank you so much for your research.  It is very insightful and I am really learning from it.

I am very confused about how the data of the patients is distributed. I understand we have 10 patients. And we have 10 folds, where each fold represents 9 training patients and 1 validation patient.

However, I cannot understand how this is done for the Domain data? We have train_feature, train_targets, val_feature, val_targets. These represents the training and validation data with their corresponding labels (5 classes).  How are train_domin, val_domin prepared?

Additionally, what data is left out to evaluate the trained model?  I believe it should be data that was never seen before during training in any of the train or val sets.

May I ask if you can briefly explain how the data is split for the training process?
Thank you so much

**Comment by ziyujia** [comment]
1. Domain labels: taking the ISRUC-S3 dataset as an example, each fold contains data from 9 training patients. The `train_domin` is set to one-hot encoding from 0 to 8 (i.e., the domain label of the first patient is [1,0,0,...,0], and the domain label of the second patient is [0,1,0, ...,0]). Since the model uses the Keras framework, the input shapes of the training and validation data need to be consistent, so `val_domin` is set as a 9-dimensional vector with all zeros (i.e., [0,0,0,...,0]). 
2. Data division: the experimental setup we use is consistent with DeepSleepNet, and for the baselines where the experimental data division is inconsistent, we reproduce them under the same rules. There have been several versions of cross-validation, and the scheme we used does not explicitly divide the validation set from the test set, i.e., we take turns to test (validate) using data from 1 subject, and finally the test results of all 10 folds are combined as the final result. We also notice that a cross-validation approach with separate validation sets would be more convincing, and we will improve this in the subsequent study. Thank you for your attention to our research!
